storage/series_test.go

```diff
@@ -40,31 +40,31 @@ func TestListSeriesIterator(t *testing.T) {
 	require.Equal(t, chunkenc.ValFloat, it.Seek(1))
 	ts, v := it.At()
 	require.Equal(t, int64(1), ts)
-	require.Equal(t, 1., v)
+	require.InDelta(t, 1., v, 0.01)
 
 	// Seek one further, next sample still has ts=1.
 	require.Equal(t, chunkenc.ValFloat, it.Next())
 	ts, v = it.At()
 	require.Equal(t, int64(1), ts)
-	require.Equal(t, 1.5, v)
+	require.InDelta(t, 1.5, v, 0.01)
```

Comment:
When something is repeated like this over and over. I would have created a testing helper function

```go
requireFloatEqual(t *testing, expected float64, actual float64) {
   t.Helper()
   
	 require.InDelta(t, expected, actual, 0.01, "msg %f %f", expected, actual)
}

requireFloatZero(t *testing, actual float64) {
   t.Helper()
   
	 requireFloatEqual(t, 0., actual)
}

requireFloatOne(t *testing, actual float64) {
   t.Helper()
   
	 requireFloatEqual(t, 1., actual)
}
```

Or anything like that

Maybe something in prom_utilstest, you could pass a counter and call ToFloat64

```go

func requireCounterZero(t *testing.T, counter prometheus.Counter) {

      t.Helper()

      requireFloatOne(t, prom_testutil.ToFloat64(counter))
}
```

This is pure pseudo code, written on a phone